dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApiMockTest.java

```diff
@@ -121,10 +121,9 @@ public void testListServersWithDatacenterFiltering() throws Exception {
       return Uris.uriBuilder("/caas/2.4/6ac1e746-b1ea-4da5-a24e-caf1a978789d/server/server");
    }
 
-   public void testListServers_404() throws Exception {
-      server.enqueue(response404());
-      assertTrue(serverApi().listServers().concat().isEmpty());
-      assertSent(HttpMethod.GET, getListServerUriBuilder().toString());
+   public void testListServers_NoServersFound() {
+      server.enqueue(emptyListResponse("server"));
```

Comment:
This test does not exercise the fallback, as it returns a 200 response. Change it to return a 400 error code (you'll need the fallbacks for the empty list mentioned previously).

Reply:
Yes, this test is testing a 200 response. This was intended as an empty list response test.
404 is not possible, as previously mentioned, hence the rewrite.

For error code 400 - there is already testDeployServerReturnsUnexpectedError
(Actually it is a preexisting bug in DimensionDataCloudControlErrorHandler that responseCode = UNEXPECTED_ERROR results in ResourceNotFoundException)
Maybe we need 400 test for list and get Server. I can add these, but I am not sure if it belongs here.

Reply:
It is OK now that there is no fallback defined for those methods.